MongoDB 3.2 Limit / Skip not working

```
{ [MongoError: Failed to parse: { find: "users", filter: {}, skip: "0", limit: "30" }. 'skip' field must be numeric.]
  name: 'MongoError',
  message: 'Failed to parse: { find: "users", filter: {}, skip: "0", limit: "30" }. \'skip\' field must be numeric.',
  waitedMS: 0,
  ok: 0,
  errmsg: 'Failed to parse: { find: "users", filter: {}, skip: "0", limit: "30" }. \'skip\' field must be numeric.',
  code: 9,
  statusCode: 400 }
```

NPM Test Result:

  1) Express Read documents limit GET /Customers?limit=1 200:

      Uncaught AssertionError: 400 == 200
      + expected - actual

      -400
      +200
      
      at Request._callback (test/integration/read.js:172:18)
      at Request.self.callback (node_modules/request/request.js:198:22)
      at Request.<anonymous> (node_modules/request/request.js:1035:10)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:962:12)
      at endReadableNT (_stream_readable.js:905:12)

  2) Express Read documents skip GET /Customers?skip=1 200:

      Uncaught AssertionError: 400 == 200
      + expected - actual

      -400
      +200
      
      at Request._callback (test/integration/read.js:189:18)
      at Request.self.callback (node_modules/request/request.js:198:22)
      at Request.<anonymous> (node_modules/request/request.js:1035:10)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:962:12)
      at endReadableNT (_stream_readable.js:905:12)

  3) Express limit GET /Customers 200 - override limit in options (query.limit < options.limit):

      Uncaught AssertionError: 400 == 200
      + expected - actual

      -400
      +200
      
      at Request._callback (test/integration/options.js:173:16)
      at Request.self.callback (node_modules/request/request.js:198:22)
      at Request.<anonymous> (node_modules/request/request.js:1035:10)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:962:12)
      at endReadableNT (_stream_readable.js:905:12)



npm ERR! Test failed.  See above for more details.

--- a/lib/middleware/prepareQuery.js
+++ b/lib/middleware/prepareQuery.js
@@ -119,6 +119,8 @@ module.exports = function (options) {
         } catch (e) {
           req._ermQueryOptions[key] = req.query[key]
         }
+      } else if (key === 'limit' || key === 'skip') {
+        req._ermQueryOptions[key] = parseInt(req.query[key], 10)
       } else {
         req._ermQueryOptions[key] = req.query[key]
       }

--- a/test/integration/delete.js
+++ b/test/integration/delete.js
@@ -102,6 +102,7 @@ module.exports = function (createFn, setup, dismantle) {
           assert.equal(res.statusCode, 204)
 
           db.models.Customer.find({}, function (err, customers) {
+            assert.ok(!err)
             assert.equal(customers.length, 2)
             customers.forEach(function (customer) {
               assert.ok(customer.name !== 'John')
@@ -201,6 +202,7 @@ module.exports = function (createFn, setup, dismantle) {
           assert.equal(res.statusCode, 204)
 
           db.models.Customer.find({}, function (err, customers) {
+            assert.ok(!err)
             assert.equal(customers.length, 2)
             customers.forEach(function (customer) {
               assert.ok(customer.name !== 'John')

--- a/test/integration/read.js
+++ b/test/integration/read.js
@@ -174,6 +174,21 @@ module.exports = function (createFn, setup, dismantle) {
           done()
         })
       })
+
+      it('GET /Customers?limit=foo 200 - evaluates to NaN', function (done) {
+        request.get({
+          url: util.format('%s/api/v1/Customers', testUrl),
+          qs: {
+            limit: 'foo'
+          },
+          json: true
+        }, function (err, res, body) {
+          assert.ok(!err)
+          assert.equal(res.statusCode, 200)
+          assert.equal(body.length, 3)
+          done()
+        })
+      })
     })
 
     describe('skip', function () {
@@ -188,6 +203,21 @@ module.exports = function (createFn, setup, dismantle) {
           assert.ok(!err)
           assert.equal(res.statusCode, 200)
           assert.equal(body.length, 2)
+          done()
+        })
+      })
+
+      it('GET /Customers?skip=foo 200 - evaluates to NaN', function (done) {
+        request.get({
+          url: util.format('%s/api/v1/Customers', testUrl),
+          qs: {
+            skip: 'foo'
+          },
+          json: true
+        }, function (err, res, body) {
+          assert.ok(!err)
+          assert.equal(res.statusCode, 200)
+          assert.equal(body.length, 3)
           done()
         })
       })

